### PR TITLE
[alerts] Add JsonRpcApiErrorRates and WebAppServicesCrashlooping to Dedicated

### PR DESCRIPTION
**Pull request by geropl**
## Description

This is the first stab at setting up alerts for Dedicated, after following [this guide](https://github.com/gitpod-io/gitpod-dedicated/tree/main/ops/telemetry/alerts).

More alerts blocked by:
 - cannot setup alerts with `warning`, yet
 - cannot setup alerts only for Dedicated, yet
/cc @kylos101 

As a result, I took the super-conservative path and only added the minimal set.

## Related Issue(s)

Fixes WEB-583

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: operations/observability/mixins/meta/rules/server.yaml [hunk 1]
```diff
@@ -46,13 +46,12 @@ spec:
 
     # Rollout alerts
     - alert: JsonRpcApiErrorRates
-      # Reasoning: the values are taken from past data
       expr: sum(rate(gitpod_server_api_calls_total{statusCode!~"2.*|3.*|4.*"}[5m])) by (cluster) / sum(rate(gitpod_server_api_calls_total[5m])) by (cluster) > 0.04
       for: 5m
       labels:
-        # sent to the team internal channel until we fine tuned it
-        severity: warning
+        severity: critical
```
Comment (geropl) [inline operations/observability/mixins/meta/rules/server.yaml:52]: We improved significantly here, and would not have triggered based on our past data.

File: operations/observability/mixins/meta/rules/server.yaml [hunk 2]
```diff
@@ -46,13 +46,12 @@ spec:
 
     # Rollout alerts
     - alert: JsonRpcApiErrorRates
-      # Reasoning: the values are taken from past data
       expr: sum(rate(gitpod_server_api_calls_total{statusCode!~"2.*|3.*|4.*"}[5m])) by (cluster) / sum(rate(gitpod_server_api_calls_total[5m])) by (cluster) > 0.04
```
Comment (easyCZ) [inline operations/observability/mixins/meta/rules/server.yaml:49]: [nit] We could flip the `statusCode` expression to `5.*` for slightly more readability

**Comment by geropl** [comment]
/unhold

Still blocked by failing build, though.

**Comment by easyCZ** [comment]
Re-run build

**Comment by kylos101** [comment]
@geropl Re: the two points you pinged me on, these are good opportunities for iteration on our observability and alert pipeline for Dedicated.

> cannot setup alerts with warning, yet

suggestion: create a related Linear issue. Can you include an example of the desired notification channel? I assume you'd want to route warnings to team specific Slack channels. :information_source: the more metrics we allow to Grafana Cloud, the more costly it is. There's also SID-536, which I'd consider a higher priority than warnings. cc: @corneliusludmann @lucasvaltl @laushinka 

> cannot setup alerts only for Dedicated, yet

suggestion: create a related Linear issue. Can you include an example of what type of alert you'd use with Dedicated, but not gitpod.io? This sounds logical, I'm just curious what the alert might look like.

**Comment by geropl** [comment]
Done [here](https://linear.app/gitpod/issue/SID-545/alerts-enable-with-severity-warning-route-to-slack-channels) and [here](https://linear.app/gitpod/issue/SID-546/alerts-enable-dedicated-only-alerts).

**Pull request merged**